daemon/cmd/daemon.go

```diff
@@ -91,6 +91,8 @@ const (
 // Daemon is the cilium daemon that is in charge of perform all necessary plumbing,
 // monitoring when a LXC starts.
 type Daemon struct {
+	lock.RWMutex // This mutex used to synchronize test setups. It is not used for implementation code.
```

Comment:
It makes sense why this is needed in the context of how the tests are written, but it makes me sad that the tests make this a requirement - in other words, code smell.

Ideally our unit tests are not dependent on one another at all. We should ask ourselves the following questions:

* what are the pieces of the daemon (or types it depends on) which are singletons in our code base? Can we remove the need for these types? One thing comes to mind to me is `ipcache` - I had a PR out a while back which tried to remove the need for this singleton, but I never got it merged.
* are there external dependencies that the unit test suite is reliant upon (e.g., kvstore?) can we create a fake KVstore for daemon tests and instead rely upon the etcd / consul test suites to test the actual functionality?

Reply:
Thanks for the feedback; this is helpful to think about going forward.

Reply:
@christarazi is @ianvernon 's comment going to be addressed? I didn't leave a review as I assumed his concerns would be addressed in this PR. I agree that having a `lock.RWMutex` in the daemon structure seems confusing. I'm not against the field addition itself, I'm worried that the only way to fix the tests is the addition of this field.

Reply:
Yes I've found a way to avoid relying on a mutex by taking a deeper look at the cause of the race. I'll push an updated version soon.

Reply:
I've found a way to avoid using a mutex: bb4712c4f9ed72fef45eef8a6d6cef34cf91dab3